src/lua/uri.lua

```diff
@@ -169,8 +178,63 @@ local function format(uri, write_password)
     return str
 end
 
+--- Encodes the given string
+-- Uses Curl function curl_easy_escape(),
+-- see description in https://curl.haxx.se/libcurl/c/curl_easy_escape.html
+-- @function url_encode
+-- @string buf - string to be encode
+-- @returns result string or nil, err
+local function url_encode(buf)
```

Comment:
Not sure that it should be part of uri library. I'd move it to http client (to be closier to curl)